crates/feldera-types/src/transport/delta_table.rs

```diff
@@ -83,17 +83,28 @@ pub struct DeltaTableReaderConfig {
 
     /// Table column that serves as an event timestamp.
     ///
-    ///
     /// When this option is specified, and `mode` is one of `snapshot` or `snapshot_and_follow`,
-    /// the snapshot of the table is ingested in the timestamp order.  This setting is required
-    /// for tables declared with the
-    /// [`LATENESS`](https://docs.feldera.com/sql/streaming#lateness-expressions) attribute
-    /// in Feldera SQL. It impacts the performance of the connector, since data must be sorted
-    /// before pushing it to the pipeline; therefore it is not recommended to use this
-    /// settings for tables without `LATENESS`.
-    // TODO: The above semantics is expensive and unnecessary.  What is really needed is to maintain
-    // the `LATENESS` guarantee of the column, for which it is sufficient to ingest rows sorted up
-    // to lateness by issuing a series of queries for lateness-width time ranges to the table.
+    /// table rows are ingested in the timestamp order, respecting the
+    /// [`LATENESS`](https://docs.feldera.com/sql/streaming#lateness-expressions)
+    /// property of the column: each ingested row has a timestamp no more than `LATENESS`
+    /// time units earlier than the most recent timestamp of any previously ingested row.
+    /// The ingestion is performed by partitioning the table into timestamp ranges of width
+    /// `LATENESS`. Each range is processed sequentially, in increasing timestamp order.
+    ///
+    /// # Example
+    ///
+    /// Consider a table with timestamp column of type `TIMESTAMP` and lateness attribute
+    /// `INTERVAL 1 DAY`. Assuming that the oldest timestamp in the table is
+    /// `2024-01-01T00:00:00``, the connector will fetch all records with timestamps
+    /// from `2024-01-01`, then all records for `2024-01-02`, `2024-01-03`, etc., until all records
+    /// in the table have been ingested.
+    ///
+    /// # Requirements
+    ///
+    /// * The timestamp column must be of a supported type: integer, `DATE`, or `TIMESTAMP`.
+    /// * The timestamp column must be declared with non-zero `LATENESS`.
```

Comment:
zero lateness makes sense too, what happens then?
the table may happen to be sorted anyway

Reply:
There's no such thing as sorted table really. We could support zero lateness by sorting each slice read from the table, but then the size of the slice is another control know the user must specify. I was trying to keep it simple.